### DISCUSSION TITLE
Scene Graph Api  khr materials variants AR doesn't work

### DISCUSSION THREAD

**Discussion by yunusguney** [google/model-viewer#2254] [Q&A]
changes made with codes called by scene graph api doesn't work.the changes are not visible anymore in ar.the model viewer is on the way but variant does not come.default state displayed.What can I do for it?

**Comment by sercandemirtas** [comment]
Hey @yunusguney Which AR are you talking about ? Quick look or scene viewer ?

**Reply by yunusguney** [reply]
Hi @sercandemirtas ,Scene viewer.Basebros  sayfasını inceledim sorularda gezerken denk gelip,güzel iş abi eline sağlık :).

**Reply by sercandemirtas** [reply]
Teşekkürler Yunus :) Can you please send your project demo link if it is deployed?

**Reply by yunusguney** [reply]
https://codepen.io/yungundev/pen/ExZXYvG demo link is here.

Change the material to blue its correct but AR not updateing.It opening default glb/gltf
![WhatsApp Image 2021-04-04 at 15 23 07 (1)](https://user-images.githubusercontent.com/77197691/113508620-ca160200-9559-11eb-97ed-8912f42bb24b.jpeg)
![WhatsApp Image 2021-04-04 at 15 23 07](https://user-images.githubusercontent.com/77197691/113508621-caae9880-9559-11eb-81ca-f684a3a64cc4.jpeg)

**Reply by elalish** [reply]
Indeed, the Scene Graph API only works in the WebXR mode of AR, as the other AR apps download their own copy, which comes from the server unaltered. This is part of what makes WebXR so powerful.

**Reply by yunusguney** [reply]
Thank you for your return @elalish .So how does this happen?And can view the change.

https://www.khronos.org/blog/live-gltf-asset-editing-in-your-browsereven-in-ar

**Reply by elalish** [reply]
Yes, it uses our default `ar-mode`, webxr. WebXR exposes ARCore inside the browser, so you're still rendering with the same JS code. This also means you don't need to wait to redownload the model. I see you haven't specified an `ar-mode`, so if you're getting Scene Viewer, it means WebXR has failed. You may want to check your console log. Keep in mind that WebXR requires HTTPS.

**Reply by yunusguney** [reply]
Thank you for clarification and information @elalish .For example this code;

https://glitch.com/edit/#!/cheddar-gabby-ulna?path=index.html%3A38%3A22

same example on demo page.Ar-mode edited as webxr.Still the feature change cannot be displayed.How can i view the attribute change of the model in webxr

**Reply by yunusguney** [reply]
@elalish  its okey !
https://glitch.com/edit/#!/tall-held-macaroni?path=index.html%3A26%3A24
this demo my project demo and ar-mode=webxr is great and working :))

Thank you very much for your help
